storage/src/main/java/tech/pegasys/teku/storage/protoarray/ProtoNode.java

```diff
@@ -105,7 +109,12 @@ public void adjustWeight(final long delta) {
       }
       weight = weight.minus(deltaAbsoluteValue);
     } else {
-      weight = weight.plus(delta);
+      if (UInt64.MAX_VALUE.minusMinZero(delta).isGreaterThan(weight)) {
+        weight = weight.plus(delta);
+      } else {
+        LOG.trace("Unable to add delta {} to weight {}", delta, weight);
+        weight = UInt64.MAX_VALUE;
+      }
```

Comment:
One option would be to add a `safePlus(..)` method on `Uint64`, similar to what we have with `safeMinus`. Something like this:

```
public UInt64 safePlus(final UInt64 other) {
    if (value != 0 && Long.compareUnsigned(other.value, MAX_VALUE.longValue() - value) > 0) {
      return UInt64.MAX_VALUE;
    }
    return plus(value, other.value);
  }
```